### PR TITLE
Redux setup

### PR DESCRIPTION
**Pull request by 29td**
### Project requirements:
- Add Redux Toolkit using npm install react-redux @reduxjs/toolkit
- structure the application using feature folder approach and use the duck pattern for your redux file.

_lastly I will configure my Redux Store and write actions and reducer for adding and removing books._

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+const ADD_BOOK = 'ADD_BOOK';
```
Comment (rloterh) [inline src/redux/books/books.js]: - [x] Kindly ensure that the value of this action type follows the [ducks pattern](https://github.com/erikras/ducks-modular-redux), which should look like:      
```javascript
 const ADD_BOOK = 'bookstore/books/ADD_BOOK';

File: src/redux/books/books.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+const REMOVE_BOOK = 'REMOVE_BOOK';
```
Comment (rloterh) [inline src/redux/books/books.js]: - [x] Kindly ensure that the value of this action type follows the [ducks pattern](https://github.com/erikras/ducks-modular-redux), which should look like:      
```javascript
 const REMOVE_BOOK = 'bookstore/books/REMOVE_BOOK';

File: src/redux/categories/categories.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+const UNDER_CONSTRUCTION = 'UNDER_CONSTRACTION';
+const COMPELET = 'COMPLETE';
```
Comment (rloterh) [inline src/redux/categories/categories.js]: - [x] Kindly ensure that the value of this action type follows the [ducks pattern](https://github.com/erikras/ducks-modular-redux), which should look like:      
```javascript
 const COMPLETE = 'bookstore/categories/COMPLETE';

File: src/redux/categories/categories.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+export const complete = () => ({
+  type: COMPELET,
+});
+
+const categoriesReducer = (state = categories, action) => {
+  switch (action.type) {
+    case UNDER_CONSTRUCTION:
+      return false;
+    case COMPELET:
```
Comment (rloterh) [inline src/redux/categories/categories.js]: - [x] This module requires that if action is of type COMPLETE (which in your case represents CHECK_STATUS), then the function should return a string `'Under construction'` else if action type is different, it should return the state with no change

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Nonhlanhla, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Highlights 👍  
- reducer  with `configureStore` properly created


## Required Changes ♻️
_Kindly check the comments under the review._


<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by 29td** [comment]
Thank you for the feedback, changes are made as requested.

**Review by prolajumokeoni — APPROVED** [review]
Hi @29td, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- No linter errors ✅ 
- Git flow was used correctly 🚀
- Good Pr title and description 💪🏿 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by 29td** [comment]
Thank you 👍

**Pull request merged**